query/sources/impl/src/main/java/org/codice/ddf/admin/sources/csw/CswSourceUtils.java

```diff
@@ -115,33 +115,31 @@ public CswSourceUtils(ConfiguratorSuite configuratorSuite) {
             taskList.add(callables);
         }
 
-        PrioritizedBatchExecutor<ReportWithResultImpl<CswSourceConfigurationField>, ReportWithResultImpl<CswSourceConfigurationField>>
+        PrioritizedBatchExecutor<Report<CswSourceConfigurationField>, Report<CswSourceConfigurationField>>
                 prioritizedExecutor = new PrioritizedBatchExecutor(THREAD_POOL_SIZE,
                 taskList,
                 new SourceTaskHandler<CswSourceConfigurationField>());
 
-        Optional<ReportWithResultImpl<CswSourceConfigurationField>> result =
-                prioritizedExecutor.getFirst();
+        Optional<Report<CswSourceConfigurationField>> result = prioritizedExecutor.getFirst();
 
         if (result.isPresent()) {
             return result.get();
         } else {
-            return new ReportWithResultImpl<CswSourceConfigurationField>().addArgumentMessage(
-                    unknownEndpointError(hostField.path()));
+            return Reports.from(unknownEndpointError(hostField.path()));
         }
     }
 
-    public ReportWithResultImpl<CswSourceConfigurationField> getCswConfigFromUrl(UrlField urlField,
+    public Report<CswSourceConfigurationField> getCswConfigFromUrl(UrlField urlField,
             CredentialsField creds) {
-        ReportWithResultImpl<ResponseField> responseResult = requestUtils.sendGetRequest(urlField,
+        Report<ResponseField> responseResult = requestUtils.sendGetRequest(urlField,
                 creds,
                 GET_CAPABILITIES_PARAMS);
 
-        if (responseResult.containsErrorMsgs()) {
-            return (ReportWithResultImpl) responseResult;
+        if (responseResult.containsErrorMessages()) {
+            return (Report) responseResult;
```

Comment:
We can return `return Reports.from(responseResult.getErrorMessages());` because it doesn't like `Reports.from(responseResult);`